src/Streamly/Internal/Data/Array/Stream/Mut/Foreign.hs

```diff
@@ -21,6 +21,8 @@ module Streamly.Internal.Data.Array.Stream.Mut.Foreign
     , compactLE
     , compactEQ
     , compactGE
+    , Array(..)
+    , partitionBy
```

Comment:
Why is this added?

Reply:
used in Streamly/Benchmark/Data/Array/Foreign/Mut/Array.hs

Reply:
Why is a mutable array API being exported via ArrayStream module and then used from there? Why aren't we using it directly from the mutable array module?

Reply:
This is the wrong module to export these from. The right module is `Data.Array.Foreign.Mut` and these are already exported from that module. So we can just remove these from here.